src/Visualization/Python/PlotCce.py

```diff
@@ -0,0 +1,278 @@
+                " '--extraction-radius'/'-r'."
+            )
+        # If we didn't specify an extraction radius, the subfile name is just
+        # the one listed in the file. If the extraction radius was specified
+        # (and we've now guaranteed there is more than one subfile) use that as
+        # the subfile name
+        cce_subfile_name = (
+            cce_subfiles[0]
+            if (extraction_radius is None or backward_cce_group is not None)
+            else f"SpectreR{extraction_radius:04}.cce"
+        )
+        cce_subfile = h5file.get(cce_subfile_name)
+        if cce_subfile is None:
+            raise click.UsageError(
+                f"Could not find Cce subfile {cce_subfile} in H5 file"
+                f" {h5_filename}. Available subfiles"
+                f" are:\n{available_subfiles(h5file, extension='cce')}"
+            )
+
+        suffix = ".dat" if backward_cce_group is not None else ""
+
+        # Only take the columns that we need and prefix the columns with their
+        # quantity so we can have just one DataFrame
+        data = pd.concat(
+            [
+                to_dataframe(cce_subfile.get(quantity + suffix))
+                .set_index("time")[modes]
+                .add_prefix(quantity)
+                for quantity in plot_quantities
+            ],
+            axis=1,
+        )
+
+    # Restrict to x-bounds if we have any
+    if x_bounds:
+        data = data[(data.index >= x_bounds[0]) & (data.index <= x_bounds[1])]
+
+    # Set up the plots
+    fig, axes = plt.subplots(len(plot_quantities), 1, sharex=True)
+    axes = list(axes)
+    # Make the legend look a bit nicer
+    divisor = 1 if (real or imag) else 2
+    num_col = min(4, len(modes) / divisor)
+    cycle = plt.rcParams["axes.prop_cycle"].by_key()["color"]
+
+    # Plot quantities and configure the axes and legend
+    for i, quantity in enumerate(plot_quantities):
+        ax = axes[i]
+        for j, mode in enumerate(modes):
+            # If we are plotting both real and imaginary modes, make real solid
+            # and imaginary dashed. If we are plotting only real or only
+            # imaginary, then the lines are solid
+            linestyle = (
+                "solid" if (real or imag or "Real" in mode) else "dashed"
+            )
+            # If we are plotting both real and imaginary modes, make the
+            # real/imag lines for the same mode the same color. If we are
+            # plotting only real or only imaginary, then just cycle regularly
+            # through the colors
+            cycle_idx = (
+                j % len(cycle) if (real or imag) else (j // 2) % len(cycle)
+            )
+            if i == 0:
+                ax.plot(
+                    data.index,
+                    data[f"{quantity}{mode}"],
+                    color=cycle[cycle_idx],
+                    linestyle=linestyle,
+                    label=mode,
```

Comment:
`label=mode if i == 0 else None,`

Or just always set the label, since you only call `legend()` on the first plot anyway.